bin/cli.js

```diff
@@ -18,22 +19,29 @@ program
 program
   .command('train')
   .description('Train your mine locally using a sonar smart contract')
-  .option('-m, --mine-address <hexstring>', 'Blockchain address for the mine to use')
+  .option('-m, --mine-address <hexstring or auto>', 'Blockchain address for the mine to use. `auto`` sets the mine to pick a random account.')
   .option('-c, --contract-address <hexstring>', 'Sonar smart contract address for the mine to use')
   .option('-i, --ipfs-url [url]', 'Url of the IPFS node (Default: "/ip4/127.0.0.1/tcp/5001")')
   .option('-e, --ethereum-url [url]', 'Url to the ethereum network to use (Default: "http://localhost:8545")')
   // TODO: Add dev mode with watching
-  .action((options) => {
-    const mineAddress = options.mineAddress
+  .action(async (options) => {
+    let mineAddress = options.mineAddress
     const contractAddress = options.contractAddress
 
     // HACK: Commander required only works with empty arguments
     if (!mineAddress) return console.log('--mine-address required')
     if (!contractAddress) return console.log('--contract-address required')
-
     const ethereumUrl = options.ethereumUrl || 'http://localhost:8545'
+
+    const web3 = new Web3(new Web3.providers.HttpProvider(ethereumUrl))
+
+    if (mineAddress === 'auto') {
+      const mineAddresses = await web3.eth.getAccounts()
+      mineAddress = mineAddresses[0]
```

Comment:
well I was thinking if we make it random we have a 80% solution to get multiple mines running. There is a chance that we might get duplicates on addresses but if we only go for 1 mine in 10 possible addresses it might work some times haha